integrations/Rockerbox/browser.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import logger from "../../utils/logUtil";
+import { NAME } from "./constants";
+import { LOAD_ORIGIN } from "../ScriptLoader";
+import { getHashFromArray } from "../utils/commonUtils";
+
+class Rockerbox {
+  constructor(config) {
+    this.clientAuthId = config.clientAuthId;
+    this.name = NAME;
+    this.customDomain = config.customDomain;
+    this.enableCookieSync = config.enableCookieSync;
+    this.eventsMap = config.eventsMap || [];
+    this.useNativeSDKToSend = config.useNativeSDKToSend;
+  }
+
+  init() {
+    logger.debug("=== In init Rockerbox ===");
+    const host = this.customDomain ? this.customDomain : "getrockerbox.com";
+    const library =
+      this.customDomain && this.enableCookieSync ? "wxyz.rb" : "wxyz.v2";
+    (function (d, RB) {
+      if (!window.RB) {
+        window.RB = RB;
+        RB.queue = RB.queue || [];
+        RB.track =
+          RB.track ||
+          function () {
+            RB.queue.push(Array.prototype.slice.call(arguments));
+          };
+        RB.initialize = function (s) {
+          RB.source = s;
+        };
+        var a = d.createElement("script");
+        a.type = "text/javascript";
+        a.async = !0;
+        a.src = `https://${host}/assets/${library}.js`;
+        a.dataset.loader = LOAD_ORIGIN;
+        const f = d.getElementsByTagName("script")[0];
+        f.parentNode.insertBefore(a, f);
+      }
+    })(document, window.RB || {});
+    window.RB.disablePushState = true;
+    window.RB.queue = [];
+    window.RB.initialize(this.clientAuthId);
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Rockerbox===");
+    return !!window.RB && !!window.RB.loaded;
+  }
+
+  isReady() {
+    logger.debug("===In isReady Rockerbox===");
+    return !!window.RB;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Rockerbox Identify===");
+    const { message } = rudderElement;
+    const { userId } = message;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+    window.RB.track("identify", {
+      external_id: userId,
+      email,
+      phone_number: message.context.traits.phone,
+    });
+  }
+
+  track(rudderElement) {
+    if (!this.useNativeSDKToSend) {
+      logger.info("The toggle is disabled. Track call will not be sent");
```

Comment:
Add more information about the toggle. It's very vague right now.

Reply:
Updated it with more details. The details of hybrid mode will be taken up in a doc